From 52c0e33aab84420e3ab53caf8f0ee5a49c704340 Mon Sep 17 00:00:00 2001
From: Lionel Debieve <lionel.debieve@foss.st.com>
Date: Fri, 25 Feb 2022 10:37:40 +0100
Subject: [PATCH 007/841] feat(makefile): add macro and variable to generate
 metadata

Add path to fwu_gen_metadata tool and associated macro to
generate the FWU metadata from platform.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git c4bbefb82bfc0d2879fdccc1e129486f287caa37 feat(makefile): add macro and variable to generate metadata]
Signed-off-by: Lionel Debieve <lionel.debieve@foss.st.com>
Change-Id: I3045af52d9374907751ab88404114d298afbe20d
---
 Makefile                     | 12 ++++++++++++
 make_helpers/build_macros.mk | 10 ++++++++++
 2 files changed, 22 insertions(+)

diff --git a/Makefile b/Makefile
index 9c4a83892..62da22ce8 100644
--- a/Makefile
+++ b/Makefile
@@ -133,6 +133,10 @@ ENCTOOL			?=	${ENCTOOLPATH}/encrypt_fw${BIN_EXT}
 FIPTOOLPATH		?=	tools/fiptool
 FIPTOOL			?=	${FIPTOOLPATH}/fiptool${BIN_EXT}
 
+# Variables for use with Firmware Update Metadata
+FWUMDTOOLPATH		?=	tools/fwu_gen_metadata/
+FWUMDTOOL		?=	${FWUMDTOOLPATH}/fwumd_tool.py
+
 # Variables for use with sptool
 SPTOOLPATH		?=	tools/sptool
 SPTOOL			?=	${SPTOOLPATH}/sptool.py
@@ -587,6 +591,14 @@ ifeq (${ENABLE_PAUTH}, 1)
 	BL_COMMON_SOURCES	+=	lib/extensions/pauth/pauth_helpers.S
 endif
 
+################################################################################
+# PSA Firmware Update arguments definition
+################################################################################
+ifeq (${PSA_FWU_SUPPORT},1)
+FWUMD_ARGS += --nb-fw-imgs ${NR_OF_FW_BANKS}
+FWUMD_ARGS += --nb-banks ${NR_OF_IMAGES_IN_FW_BANK}
+endif
+
 ################################################################################
 # Include the platform specific Makefile after the SPD Makefile (the platform
 # makefile may use all previous definitions in this file)
diff --git a/make_helpers/build_macros.mk b/make_helpers/build_macros.mk
index 71cf18b05..0d75618ea 100644
--- a/make_helpers/build_macros.mk
+++ b/make_helpers/build_macros.mk
@@ -163,6 +163,16 @@ $(2): $(1) enctool
 	$$(Q)$$(ENCTOOL) $$(ENC_ARGS) -i $$< -o $$@
 endef
 
+# GEN_METADATA
+define GEN_METADATA
+$(dir $(2)):
+	$$(Q)mkdir -p $$(dir $(2))
+
+$(2): $(1) | $(dir $(2))
+	$$(ECHO) "  GEN_METADATA     $$<"
+	$$(Q)$$(FWUMDTOOL) $$(FWUMD_ARGS) jsonparse $$< -b $$@
+endef
+
 # TOOL_ADD_PAYLOAD appends the command line arguments required by fiptool to
 # package a new payload and/or by cert_create to generate certificate.
 # Optionally, it adds the dependency on this payload
-- 
2.47.2

